Spring-03

Max-Pochepets opened the pull request:
Spring MVC

- Spring MVC configured
- UserResponseDTO created
- UserMapper created
- UserController implemented

Max-Pochepets committed:
Initial

Max-Pochepets committed:
descriptions deleted in db.properties

Max-Pochepets committed:
removed UserMapper's userDto field

Max-Pochepets committed:
JSP and JSTL dependencies removed

dmytro-hryhoruk commented on src/main/java/com/booking/service/controller/UserController.java:
```diff
@@ -0,0 +1,61 @@
+                    break;
+                }
+                default: break;
+            }
```
It would take less place to just create four users

dmytro-hryhoruk commented on src/main/java/com/booking/service/dao/impl/UserDaoImpl.java:
```diff
@@ -38,6 +39,16 @@ public void add(User user) {
+    public Optional<User> get(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            User user = session.get(User.class, id);
+            return Optional.ofNullable(user);
```
let's do this inline

dmytro-hryhoruk commented on src/main/java/com/booking/service/dto/UserResponseDto.java:
```diff
@@ -0,0 +1,25 @@
+
+@Component
+public class UserResponseDto {
+    private String name;
```
let's add id too

dmytro-hryhoruk commented on src/main/java/com/booking/service/service/UserMapper.java:
```diff
@@ -0,0 +1,22 @@
+        return userDto;
+    }
+
+    public User fromDto(UserResponseDto inputUserDto) {
```
we don't need this method for now

dmytro-hryhoruk commented on src/main/java/com/booking/service/service/impl/UserServiceImpl.java:
```diff
@@ -19,6 +19,14 @@ public void add(User user) {
 
+    @Override
+    public User get(Long id) {
+        if (userDao.get(id).isEmpty()) {
+            return null;
+        }
+        return userDao.get(id).get();
```
is it okay to make two requests to DB for one user?

Max-Pochepets committed:
Refactor:

inject() simplified
UserDao.get() simplified
added id for UserResponseDto
fromDto() method removed
optionalUser variable created to minimize quantity of requests to DB

dmytro-hryhoruk commented on src/main/java/com/booking/service/service/impl/UserServiceImpl.java:
```diff
@@ -19,6 +20,15 @@ public void add(User user) {
 
+    @Override
+    public User get(Long id) {
+        Optional<User> optionalUser = userDao.get(id);
+        if (optionalUser.isEmpty()) {
+            return null;
+        }
+        return optionalUser.get();
```
1) you can do this check inline with optional;
2) It's better to throw an exception(you can use orElseThrow for this)

Max-Pochepets committed:
Refactor:

UserServiceImpl.get() logic refactored
